### DISCUSSION TITLE
Can we consider switching framer-motion to lazy loading?

### DISCUSSION THREAD

**Discussion by u3u** [heroui-inc/heroui#1539] [Feedback]
Maybe we can put `LazyMotion` in `NextUIProvider`
https://www.framer.com/motion/guide-reduce-bundle-size/#lazy-loading

**Comment by jrgarciadev** [comment]
Hey @u3u yes we considered that, but Is not that easy because not all components have `framer-motion` as peer dependency and NextUI also supports individual package installations, so we cannot just add `LazyMotion`to the `NextUIProvider` due to not all components will use it, we're thinking in a better solution 👍🏻

**Reply by u3u** [reply]
Indeed, I did not consider the situation of installing components separately. Thank you for your work!

**Reply by wangyq0313** [reply]
> 嘿@u3u是的，我们考虑过这一点，但这并不容易，因为并非所有组件都具有`framer-motion`对等依赖关系，而且 NextUI 还支持单独的包安装，所以我们不能仅仅因为并非所有组件都会使用它而添加`LazyMotion`它`NextUIProvider`，我们正在考虑一个更好的解决方案👍🏻

I hope this problem can be solved

**Reply by wingkwong** [reply]
@wangyq0313 `LazyMotion` has been introduced in individual components since 2.3.0.

**Reply by wangyq0313** [reply]
> @wangyq0313 `LazyMotion`自 2.3.0 版本以来，一直在这个组件中引入。

Thank you, I am updating now
